### PR TITLE
Onboarding a11y fixes

### PR DESCRIPTION
**Pull request by jacobherrington**
## What type of PR is this? (check all applicable)

- [ ] Refactor
- [ ] Feature
- [x] Bug Fix
- [ ] Documentation Update

## Description
This fixes some small complaints from aXe, as one of the steps in refactoring onboarding.

There are a handful of more involved a11y fixes we could make, but I pulled these into a separate PR because they should be easy to merge.

## Added to documentation?

- [ ] docs.dev.to
- [ ] readme
- [x] no documentation needed

## [optional] What gif best describes this PR or how it makes you feel?

Fixing a car!

### CODE REVIEW

File: app/views/layouts/application.html.erb [hunk 1]
```diff
@@ -53,6 +53,7 @@
         <%= javascript_include_tag "application" %>
       <% end %>
       <%= yield(:page_meta) %>
+      <%# TODO Remove the zoom restriction, this is pretty bad for a11y %>
       <meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=no">
```
Comment (jacobherrington) [inline app/views/layouts/application.html.erb]: @benhalpern Do you happen to know why this is necessary? I feel like we should be able to remove it, but I didn't want to recreate a bug that this is hiding.

Comment (benhalpern) [reply app/views/layouts/application.html.erb]: Yes, it was originally in place to hide a bug and also to make the app more "app-like" for the mobile-wrapped context.

But I'm not even 100% sure the bug is even still present (Basically I think the width of the page is over 100% and this is kind of hiding that issue).

I'd rather fix this than hold onto the constraint due to a bug. @rhymes actually has an open PR from a while back where we discuss this.

https://github.com/thepracticaldev/dev.to/pull/2150

I want to get this fixed. I'm not even sure the old bug even still exists and I'm happy to forget about the native mobile parity concern in favor of a11y.

Comment (benhalpern) [reply app/views/layouts/application.html.erb]: Yes, it was originally in place to hide a bug and also to make the app more "app-like" for the mobile-wrapped context.

But I'm not even 100% sure the bug is even still present (Basically I think the width of the page is over 100% and this is kind of hiding that issue).

I'd rather fix this than hold onto the constraint due to a bug. @rhymes actually has an open PR from a while back where we discuss this.

https://github.com/thepracticaldev/dev.to/pull/2150

I want to get this fixed. I'm not even sure the old bug even still exists and I'm happy to forget about the native mobile parity concern in favor of a11y.

Comment (jacobherrington) [reply app/views/layouts/application.html.erb]: Cool, thanks @benhalpern. I'll make a separate PR for that issue after I catch up on #2150.

Comment (rhymes) [reply app/views/layouts/application.html.erb]: +1 on fixing it, I think the argument was there was, maybe, an issue in the iOS native app

File: app/views/pages/_terms_text.html.erb [hunk 2]
```diff
@@ -1,4 +1,4 @@
-    <h3 id="terms">
+    <h3 id="terms-text">
```
Comment (rhymes) [inline app/views/pages/_terms_text.html.erb]: doesn't this change anything?

Comment (jacobherrington) [reply app/views/pages/_terms_text.html.erb]: There were two elements with this Id on the page. :sweat_smile:

Comment (nickytonline) [reply app/views/pages/_terms_text.html.erb]: Are the IDs used for anything? It's not form data as they're `<h3 />` tags, so my guess is they aren't being used, unless they're being used with hashbangs, e.g. https://dev.to/somePage#terms

If they serve no purpose, I would just remove them.

**Pull request merged**